backend/allNonVisualChecksHavePassed.ts

```diff
@@ -0,0 +1,15 @@
+import { getOctokit } from './getOctokit';
+
+export const allNonVisualChecksHavePassed = async (owner: string, repo: string, sha: string): Promise<boolean> => {
+  const octokit = getOctokit(owner, repo);
+
+  const { data } = await octokit.rest.checks.listForRef({
+    owner,
+    repo,
+    ref: sha
+  });
+  return data.check_runs.filter(({ name }) => !isVisualTest(name)).every(checkRun => checkRun.conclusion === 'success');
```

Comment:
i think we want might want to be more specific in this check... particularly regarding not counting `skipped` as a failure for non-required workflows triggered on label adds

```suggestion
  return data.check_runs.filter(({ name }) => !isVisualTest(name)).every(checkRun => checkRun.conclusion === 'success' || checkRun.conclusion === 'skipped');
```